### ISSUE TITLE
How to make Exercism even more useful for newcomers

### ISSUE THREAD

**Issue by m-a-ge** [exercism/discussions#185]
### Issues
As a newcomer I would like to give you my feedback of using Exercism. 

First of all, it's really unique and helpful resource for learning a new language syntax by solving well thought exercises.  But I think it lacks some support for learners when it comes to people who are not that familiar with track`s language syntax or best practices adopted for that language and they face 'fear of a blank sheet of paper' (blank text file in our case) and don't know what to do next. 
They might not be familiar with a concept of TDD at all etc. I didn't have such problems since I have experience working with several languages (JavaScript, Java, Groovy, Python and others) and I'm familiar with git and command line.

Other issue is that there are no (or very little) 'practical` exercises which teach you how to (also relevant to me):
+ use standard language library
+ work with `csv`, `xml`, `json` etc
+ work with `HTTP`
+ writing\reading files
+ working with APIs
+ etc

I understand it'll be hard to add them because it's almost impossible to test user's solutions for some of them. But if we manage to add such exercises it will make Exercism very successful since it will not only help you get used to language syntax and its correct usage, it will also show you how to use the language in real world applications. Exercism will be priceless in that case!

In essence, I would like to be able to decide if the language is good for me to use in day-to-day programming after completing the chosen track, to understand not only language syntax, but also its ecosystem, build tools, third-party libraries etc.

### Possible Solutions
I believe some of the issues mentioned above have been already addressed in ver. 2.0 of Exercism - https://github.com/exercism/v2-feedback. 

As for others I can suggest the following:
- maybe it's a good idea to somehow show links to suggested reading materials or videos (preferred) before the user starts working on the exercise. This materials may explain language syntax with examples, best practices etc. `hints.md` might be used for that
- divide exercise in sections like:
  - core syntax
  - core standard library
  - core best practices 
  - real world applications
  - ...
- think of how we can add 'practical exercises' which will help in understanding how the language can be used in day-to-day programming

What do you think?

**Comment by kytrinyx** [comment]
Thank you @m-a-ge!

> But I think it lacks some support for learners when it comes to people who are not that familiar with track`s language syntax or best practices adopted for that language

Yes, I absolutely agree. One of our goals once we have the new site running is to focus on finding (or creating, if missing) the resources that will help support new learners.

> Other issue is that there are no (or very little) 'practical` exercises which teach you how to [...]

So this is where it gets interesting. I agree on most of your points here, but not entirely all of them.

In our work on the v2 prototype we did a lot of thinking about what our goals are for Exercism. You can read our summary of our thinking here: https://github.com/exercism/docs/blob/master/about/conception/goals.md
and read about the conclusion here:
https://github.com/exercism/docs/blob/master/about/goal-of-exercism.md

You'll notice that we deliberately cut this off before we introduce real world applications, APIs, and third party libraries.

What we do hope, is that with the new website the maintainers of each track will consider the track's curriculum as a whole and really think about:

- what are the core language concepts?
- what are the different parts of the standard library?
- what are the best practices?

Then for each question, evaluate whether or not the track has exercises that introduce it, and exercises to help practice it.

I think manipulating files, CSV, JSON, etc, are all good candidates for Exercism exercises. Many languages have CSV and JSON and file io as part of their standard libraries.

Some languages even have HTTP as part of their standard libraries. In Go it would make a lot of sense to have an exercise that teaches you to use HTTP for that reason. Also it has a very nice package in the standard library that lets you set up a fake test server that your code can test against, in just a few lines of code.

In conclusion: You're right. We can do so much more to help prepare people. Much of this work will be about finding and/or creating great resources, and much will be about improving the curriculum of each individual track from the point of view of _what are the important basics in this language_?

**Comment by m-a-ge** [comment]
@kytrinyx, thanks for clarification
